temperature/services.py

```
@@ -0,0 +1,26 @@
+import os
+
+import httpx
+
+from city.models import CityModel
+from dotenv import load_dotenv
+
+load_dotenv()
+
+API_KEY = os.getenv("API_KEY")
```

This line retrieves the `API_KEY` but doesn't check if it was actually found. If the environment variable isn't set, `API_KEY` will be `None`, and the external API call will fail with a `401 Unauthorized` error. It's better practice to add a check here to ensure the `API_KEY` has a value and raise a configuration error if it's missing. This helps the application fail fast with a clear, descriptive error message.